xcm/xcm-builder/src/pay.rs

```diff
@@ -0,0 +1,75 @@
+// Copyright Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! `PayOverXcm` struct for paying through XCM and getting the status back
+
+use frame_support::traits::{
+	tokens::{Pay, PaymentStatus},
+	Get,
+};
+use sp_std::{marker::PhantomData, vec};
+use xcm::prelude::*;
+
+pub struct PayOverXcm<DestChain, Router, Querier, BlockNumber>(
```

Comment:
Imaging if you are a parachain team wanting to implement this feature. Where will you put this file? This isn't any core xcm feature and I am not sure if this is the right place for it.

Reply:
It is a configurable XCM item, and I'm not sure where else you'd put it. Why would you think that anything goes into XCM builder MUST be a core XCM feature? That is certainly not what the crate description says.